web/app/js/components/ResourceDetail.jsx

```diff
@@ -122,6 +129,16 @@ export class ResourceDetailBase extends React.Component {
     }, []);
   }
 
+  startServerPolling() {
+    this.loadFromServer();
+    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+  }
+
+  stopServerPolling() {
```

Comment:
wherever we add `stopServerPolling`, we need to make sure we reset the state to set `this.state.pendingRequests` to be false. Otherwise, if we navigate away from a page while a request is pending (i.e. pendingRequests is true) even if we re-focus the page after a while, the current requests will have been cancelled,  but `pendingRequests` will still be true (and will never get set to false again, so the page will never re-poll.

I used the logging in this diff to highlight this bug (note that we never reach line 150 because pendingRequests is always true):
![Screen Shot 2019-12-09 at 6 06 27 PM](https://user-images.githubusercontent.com/549258/70489040-67b90880-1aaf-11ea-9d82-3b976daf6e20.png)

```diff
diff --git a/web/app/js/components/ResourceDetail.jsx b/web/app/js/components/ResourceDetail.jsx
index 6ba4742c..503a5f47 100644
--- a/web/app/js/components/ResourceDetail.jsx
+++ b/web/app/js/components/ResourceDetail.jsx
@@ -130,21 +130,24 @@ export class ResourceDetailBase extends React.Component {
   }

   startServerPolling() {
+    console.log("startServerPolling");
     this.loadFromServer();
     this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
   }

   stopServerPolling() {
+    console.log("stopServerPolling");
     window.clearInterval(this.timerId);
     this.api.cancelCurrentRequests();
   }

   loadFromServer() {
+    console.log("145 loadFromServer");
     if (this.state.pendingRequests) {
       return; // don't make more requests if the ones we sent haven't completed
     }
     this.setState({ pendingRequests: true });
-
+    console.log("150 loadFromServer");
     let { resource } = this.state;

     let apiRequests =
@@ -176,7 +179,7 @@ export class ResourceDetailBase extends React.Component {
     }

     this.api.setCurrentRequests(apiRequests);
-
+    console.log("sending requests...", this.api.getCurrentPromises().length);
     Promise.all(this.api.getCurrentPromises())
       .then(([resourceRsp, podListRsp, podMetricsRsp, upstreamRsp, downstreamRsp, edgesRsp]) => {
         let resourceMetrics = processSingleResourceRollup(resourceRsp, resource.type);
@@ -235,7 +238,7 @@ export class ResourceDetailBase extends React.Component {
         if (hasHttp || hasTcp) {
           lastMetricReceivedTime = Date.now();
         }
-
+        console.log("setting state...");
         this.setState({
           resourceMetrics,
           resourceIsMeshed,
```